src/warc2zim/content_rewriting/js.py

```diff
@@ -0,0 +1,252 @@
+import re
+from typing import Any, Callable, Optional, Iterable, Tuple, Mapping, List
+
+TransformationRule = Tuple[re.Pattern, Callable[[re.Match], str]]
+
+
+IMPORT_RX = re.compile(r"""^\s*?import\s*?[{"'*]""")
+EXPORT_RX = re.compile(
+    r"^\s*?export\s*?({([\s\w,$\n]+?)}[\s;]*|default|class)\s+", re.M
+)
+
+IMPORT_MATCH_RX = re.compile(
+    r"""^\s*?import(?:['"\s]*(?:[\w*${}\s,]+from\s*)?['"\s]?['"\s])(?:.*?)['"\s]""",
+)
+
+IMPORT_HTTP_RX = re.compile(
+    r"""(import(?:['"\s]*(?:[\w*${}\s,]+from\s*)?['"\s]?['"\s]))((?:https?|[./]).*?)(['"\s])""",
+)
+
+GLOBAL_OVERRIDES = [
+    "window",
+    "globalThis",
+    "self",
+    "document",
+    "location",
+    "top",
+    "parent",
+    "frames",
+    "opener",
+]
+
+GLOBALS_CONCAT_STR = "|".join(
```

Comment:
Not sure it has value on its own. Given it's already hardly readable, I think it can be combined with `GLOBALS_RX` directly